tools/test_bindings.py

```diff
@@ -0,0 +1,147 @@
+
+        if "bindings" in struct_name:
+            ret[struct_name] = {'size': size, 'alignment': alignment}
+
+    return ret
+
+def pahole(binary: str) -> str:
+    """Runs pahole on a binary and returns its output as a str
+
+    If pahole fails this will raise a `CalledProcessError`
+
+    @param binary: binary to run pahole on
+    @return: On success, it will return the stdout of the pahole process
+    """
+    result = subprocess.run(['pahole', binary], stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                            check=True)
+    return result.stdout
+
+def check_pahole_mismatches(old: str, new: str) -> bool:
+    """Checks for pahole mismatches in pahole information between two binaries
+
+    @param old: old Firecracker binary
+    @param new: new Firecracker binary
+    @return: false if no mismatches found, true otherwise
+    """
+    pahole_structs_1 = parse_pahole(pahole(old))
+    pahole_structs_2 = parse_pahole(pahole(new))
+
+    # We go through all the structs existing in the old firecracker binary and check for mismatches
+    # in the new one.
+    for name, prop_1 in pahole_structs_1.items():
+        # Note that the reverse, i.e. a name existing in the new binary but not in the old binary,
+        # is not a problem. That would mean we are making use of some new struct from
+        # bindgen-generated code. That does not break ABI compatibility.
+        if name not in pahole_structs_2:
+            log.warning("struct '%s' does not exist in new binary", name)
+            continue
+
+        prop_2 = pahole_structs_2[name]
+        # Size mismatches are hard errors
+        if prop_1['size'] != prop_2['size']:
+            log.error("size of '%s' does not match in two binaries", name)
+            log.error("old: %s", prop_1['size'])
+            log.error("new: %s", prop_2['size'])
+            return True
+
+        # Alignment mismatches just cause warnings
+        if prop_1['alignment'] != prop_2['alignment']:
+            log.warning("alignment of '%s' does not match in two binaries", name)
+            log.warning("old: %s", prop_1['alignment'])
+            log.warning("new: %s", prop_2['alignment'])
+        else:
+            log.info("struct '%s' matches", name)
+
+    return False
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Check bindings ABI compatibility for Firecracker')
+    parser.add_argument("firecracker_old", type=str, metavar="old-firecracker-binary",
+                        help="Firecracker binary with old bindings")
+    parser.add_argument("firecracker_new", type=str, metavar="new-firecracker-binary",
+                        help="Firecracker binary with new bindings")
+    args = parser.parse_args()
+
+    if check_pahole_mismatches(args.firecracker_old, args.firecracker_new):
+        log.error("pahole information mismatch")
+        sys.exit(1)
+    else:
+        log.info("pahole information matches")
+
+    sys.exit(0)
```

Comment:
Could be simplified:
```suggestion
    if check_pahole_mismatches(args.firecracker_old, args.firecracker_new):
        log.error("pahole information mismatch")
        sys.exit(1)
    else:
        log.info("pahole information matches")
```
Also may be worth using `raise Exception("pahole infomation mismatch")` to replace:
```python
log.error("pahole information mismatch")
sys.exit(1)
```
But I'm not sure here.